bundles/org.openhab.binding.lgthinq/src/main/java/org/openhab/binding/lgthinq/internal/handler/LGThinQBridgeHandler.java

```diff
@@ -0,0 +1,375 @@
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.lgthinq.internal.LGThinQBridgeConfiguration;
+import org.openhab.binding.lgthinq.internal.discovery.LGThinqDiscoveryService;
+import org.openhab.binding.lgthinq.lgservices.LGThinQApiClientServiceFactory;
+import org.openhab.binding.lgthinq.lgservices.LGThinQApiClientServiceFactory.LGThinQGeneralApiClientService;
+import org.openhab.binding.lgthinq.lgservices.api.TokenManager;
+import org.openhab.binding.lgthinq.lgservices.errors.LGThinqException;
+import org.openhab.binding.lgthinq.lgservices.errors.RefreshTokenException;
+import org.openhab.binding.lgthinq.lgservices.model.CapabilityDefinition;
+import org.openhab.binding.lgthinq.lgservices.model.LGDevice;
+import org.openhab.binding.lgthinq.lgservices.model.SnapshotDefinition;
+import org.openhab.core.config.core.status.ConfigStatusMessage;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.ConfigStatusBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandlerService;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link LGThinQBridgeHandler} - connect to the LG Account and get information about the user and registered
+ * devices of that user.
+ *
+ * @author Nemer Daud - Initial contribution
+ */
+@NonNullByDefault
+public class LGThinQBridgeHandler extends ConfigStatusBridgeHandler implements LGThinQBridge {
+
+    private final Map<String, LGThinQAbstractDeviceHandler<? extends CapabilityDefinition, ? extends SnapshotDefinition>> lGDeviceRegister = new ConcurrentHashMap<>();
+    private final Map<String, LGDevice> lastDevicesDiscovered = new ConcurrentHashMap<>();
+
+    private static final LGThinqDiscoveryService DUMMY_DISCOVERY_SERVICE = new LGThinqDiscoveryService();
+    static {
+        var logger = LoggerFactory.getLogger(LGThinQBridgeHandler.class);
+        try {
+            File directory = new File(THINQ_USER_DATA_FOLDER);
+            if (!directory.exists()) {
+                if (!directory.mkdir()) {
+                    throw new LGThinqException("Can't create directory for userdata thinq");
+                }
+            }
+        } catch (Exception e) {
+            logger.warn("Unable to setup thinq userdata directory: {}", e.getMessage());
+        }
+    }
+    private final Logger logger = LoggerFactory.getLogger(LGThinQBridgeHandler.class);
+    private LGThinQBridgeConfiguration lgthinqConfig = new LGThinQBridgeConfiguration();
+    private final TokenManager tokenManager;
+    private LGThinqDiscoveryService discoveryService = DUMMY_DISCOVERY_SERVICE;
+    private final @Nullable LGThinQGeneralApiClientService lgApiClient;
```

Comment:
It does not seem to be nullable.
```suggestion
    private final LGThinQGeneralApiClientService lgApiClient;
```